rpc/src/optimistically_confirmed_bank_tracker.rs

```diff
@@ -296,6 +297,9 @@ impl OptimisticallyConfirmedBankTracker {
                         *highest_confirmed_slot = slot;
                     }
                     drop(w_optimistically_confirmed_bank);
+
+                    // finalize block's minimum prioritization fee cache for this bank
+                    prioritization_fee_cache.finalize_priority_fee(slot, bank_id);
```

Comment:
finalize `prioritization_fee_cache` with OC-ed bank_id, purge fee from other banks from cache.

Reply:
hmm, actually, i think `.finalize_priority_fee()` should be called just after `Self::notify_slot_status(..., SlotNotification::OptimisticallyConfirmed)`-ing inside `OptimisticallyConfirmedBankTracker::notify_or_defer()`.

otherwise, we can't ensure bank is frozen (prioritization cache can potentially have updates until frozen) and all parent banks' prioritization fee caches are correctly finalized without skipping?

note that `BankNotification::OptimisticallyConfirmed(slot)` can have gaps and could be send before locally replaying..

so, #32692 had already bugs, it seems...

Reply:
Mmm, not familiar enough with this part of code. Let me understand how it flows first.

Reply:
yea, the gap between OptimisticallyConfirmed and bank frozen is real. Thanks for pointing to the correct solution. I opened a separate PR #33100 to address this bug.